### DISCUSSION TITLE
How can i Change shader settings?

### DISCUSSION THREAD

**Discussion by AryanShetty** [IrisShaders/Iris#397] [Q&A] [answered]
Hiya! I'm so incredibly happy that i can finally use shaders with sodium and I cannot thank the author(s) of this mod enough. I'd like help with some small things though. How does one go about changing the shader settings? I use BSL and Sildurs and i hate how both of them look with bloom and fog. How can I disable fog and bloom? I've looked about the settings and found nothing. Help would be greatly appreciated! Thanks again :)

**Comment by ghost** [comment]
Same here! I saw someone using that fancy shaders settings menu, but now I can't find that person

**Comment by ghost** [comment]
> Hiya! I'm so incredibly happy that i can finally use shaders with sodium and I cannot thank the author(s) of this mod enough. I'd like help with some small things though. How does one go about changing the shader settings? I use BSL and Sildurs and i hate how both of them look with bloom and fog. How can I disable fog and bloom? I've looked about the settings and found nothing. Help would be greatly appreciated! Thanks again :)

As of now, you can't change shader settings. Here a mod which could help though.

No Fog: https://www.curseforge.com/minecraft/mc-mods/nofog

**Reply by AryanShetty** [reply]
Thanks a lot Expalsive!! Any suggestions on what i can do about the bloom? I don't mind the fog as much as I hate the goddamn bloom. It makes the whole thing so much worse. Help is Appreciated!

**Reply by Walii22** [reply]
Hello, what you can try to do,  is go to a forge version of minecraft. Install your shaders there. Edit them and once you are done changing the settings of your shaders. Just copy and paste the shader and it's text file over to Fabric. Hope this helps. It helped me change a few settings. but not all. Good luck.

**Reply by AryanShetty** [reply]
Are you sure that works? I thought the settings would only change in the renderer since the files are in a zip ( it would be much more complicated to save changes to a zip). I'll try it out still but I'm just trying to understand why it works. Thanks a lot!

**Reply by Walii22** [reply]
Hey man It doesn’t work :(. I retested it and nothing changed sadly. At first I thought it changed, but I tried it again and nope :/. Sorry

**Reply by AryanShetty** [reply]
Haha No problem buddy! You went through all that trying to help me and thats more than enough, also why would you apologise? Thanks a lot!! But you did give me an idea, maybe i can look into the shader pack files and change the settings over there. Thanks a lot again :)

**Comment by WereWolfix** [comment, marked as answer]
> Hiya! I'm so incredibly happy that i can finally use shaders with sodium and I cannot thank the author(s) of this mod enough. I'd like help with some small things though. How does one go about changing the shader settings? I use BSL and Sildurs and i hate how both of them look with bloom and fog. How can I disable fog and bloom? I've looked about the settings and found nothing. Help would be greatly appreciated! Thanks again :)

I have found a way to change the settings from the shaders files. Obviously you must have unzipped the zip file. I give an example with BSL_shader. Go inside the BSL / shaders / lib folder, inside you will find a file called setting.glsl, open it with a text editor program. All the settings preceded by // will be disabled by default so if you want to have them active you have to delete the // same thing vice versa if you want to disable them add the // in front of the setting

**Comment by MqcCheeze** [comment]
Yes, an easily accessible config needs to be added. Lots of lag from Mqc

**Comment by XIYU11864** [comment]
Try this. https://github.com/HyperCubeMC/Iris
